pkg/apis/install.relay.sh/v1alpha1/relaycore_types.go

```diff
@@ -220,6 +278,16 @@ type MetadataAPIConfig struct {
 	// +optional
 	URL *string `json:"url,omitempty"`
 
+	// LogServiceEnabled defines whether the log service is enabled or not
+	//
+	// +optional
+	LogServiceEnabled bool `json:"logServiceEnabled,omitempty"`
+
+	// LogServiceURL is the URL of the service used to persist log messages.
+	//
+	// +optional
+	LogServiceURL string `json:"logServiceURL,omitempty"`
```

Comment:
Something I've been doing to slim down the api and simplify the reconciliation loops is making field's like this a pointer, then by virtue of this URL being set, the log service being enabled is implied. That means you can remove the `LogServiceEnabled` field. I then document the field saying that if it's set, that feature is enabled: https://github.com/puppetlabs/relay-core/blob/development/pkg/apis/install.relay.sh/v1alpha1/relaycore_types.go#L157-L162

Another thing I've been doing for tightly couples services like this is just being super opinionated with the DNS. The installer prefixes all the objects and knows how to derive DNS from the resource metadata (like the service name), so if you don't actually set this up with external dns or anything, you might just be able to remove these fields completely and then in the reconciler, you would check if `RelayCore.Spec.LogService != nil`.

Just a couple tips, but you know better than me for the PLS implementation.

Reply:
The `LogServiceEnabled` flag is handling a specific use-case that can't be entirely solved by use a pointer for the `LogServiceURL`. But I believe we can handle it by cross-checking against the `RelayCore.Spec.LogService` in a few places to accomplish this. I was trying to avoid directly referencing the LogService spec or state in the MetadataAPI state to reduce internal interdependence, but I don't suppose there's any real reason to avoid it here.